src/Chapter2.hs

```diff
@@ -605,7 +620,8 @@ Implement a function that duplicates each element of the list
 
 -}
 duplicate :: [a] -> [a]
-duplicate = error "duplicate: Not implemented!"
+duplicate [] = []
+duplicate (head':tail') = head' : head' : duplicate tail'
```

Comment:
Very nice solution! By standard naming conventions in Haskell, head and tail of a list are usually called `x` and `xs`. The code looks a bit cleaner and shorter with them. Also, `head` and `tail` are standard functions in Haskell, so you need to add quotes `'` (as you did) to avoid name conflicts, which is a bit awkward.

I see that short names can be not so helpful in some cases, but for such polymorphic functions they are actually fine. And after some time you get used to such notation :slightly_smiling_face: 

```suggestion
duplicate (x:xs) = x : x : duplicate xs
```

Reply:
Thank you a lot for your review

Reply:
I used `head` and `tail` in `python` code in past